Variable undeclared in assert statement with rlock/lock expr when building generated c code

### Describe the bug

When compiling the generated c code, cc complains that `error: '_t3' undeclared`.
# The test case
```shell
>v assert_lock_expr_undefined_var.v
================== C compilation error (from tcc): ==============
cc: C:/Users/adm/AppData/Local/Temp/v_0/assert_lock_expr_undefined_var.01K8AKBPNCGC11949JH1221CDM.tmp.c:8356: warning: assignment from incompatible pointer type
cc: C:/Users/adm/AppData/Local/Temp/v_0/assert_lock_expr_undefined_var.01K8AKBPNCGC11949JH1221CDM.tmp.c:8408: error: '_t3' undeclared
...
cc: C:/Users/adm/AppData/Local/Temp/v_0/assert_lock_expr_undefined_var.01K8AKBPNCGC11949JH1221CDM.tmp.c:8408: error: '_t3' undeclared
(note: the original output was 13 lines long; it was truncated to its first 2 lines + the last line)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

# The test source code
assert_lock_expr_undefined_var.v
```v
fn main() {
    println('init')
    shared c := Counter{1}
    
    println('test#1')
    assert rlock c{ c.val() } == 1
    if rlock c{ c.val() } == 1 {
        println('test#1: ok');
    }
    
    println('inc()')
    c.inc()
    
    println('test#2')
    assert rlock c{ c.val() } == 2
    if rlock c{ c.val() } == 2 {
        println('test#2: ok');
    }
    
    println('ok.')
}

struct Counter {
mut:
    value int
}

fn (shared c Counter) inc() {
    println('inc(): locking ..')
    lock c {
        println('inc(): locked')
        c.value += 1
        println('inc(): updated')
    }
    println('inc(): unlocked')
}

fn (c Counter) val() int {
    return c.value
}
```

### Reproduction Steps

```shell
>v assert_lock_expr_undefined_var.v
```

### Expected Behavior

Compiling pass

### Current Behavior

error: '_t3' undeclared

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.12 0db62d5

### Environment details (OS name and version, etc.)

Windows 10 x64
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/assert.v
+++ b/vlib/v/gen/c/assert.v
@@ -100,6 +100,9 @@ fn (mut g Gen) assert_subexpression_to_ctemp(expr ast.Expr, expr_type ast.Type) 
 				return g.new_ctemp_var_then_gen(expr, expr_type)
 			}
 		}
+		ast.LockExpr {
+			return g.new_ctemp_var_then_gen(expr, expr_type)
+		}
 		else {}
 	}
 	return unsupported_ctemp_assert_transform

--- a/vlib/v/tests/concurrency/shared_lock_expr_assert_test.v
+++ b/vlib/v/tests/concurrency/shared_lock_expr_assert_test.v
@@ -1,0 +1,38 @@
+struct Counter {
+mut:
+	value int
+}
+
+fn (shared c Counter) inc() {
+	lock c {
+		c.value += 1
+	}
+}
+
+fn (c Counter) val() int {
+	return c.value
+}
+
+fn test_main() {
+	shared c := Counter{1}
+
+	assert rlock c {
+		c.val()
+	} == 1
+	if rlock c {
+		c.val()
+	} == 1 {
+		assert true
+	}
+
+	c.inc()
+
+	assert rlock c {
+		c.val()
+	} == 2
+	if rlock c {
+		c.val()
+	} == 2 {
+		assert true
+	}
+}

